frontend/src/modules/progresschart/components/Canvas.js

```diff
@@ -0,0 +1,116 @@
+/* @flow */
+
+import * as React from 'react';
+
+import './Canvas.css';
+
+import type { Stats } from 'core/stats';
+
+
+type Props = {|
+    stats: Stats,
+    height: number,
+    width: number,
+    class: string,
+|};
+
+
+/**
+ * Render current resource progress data on canvas.
+ */
+export default class Canvas extends React.Component<Props> {
```

Comment:
Thinking about this a bit more, I think this should be called `ProgressChart` because that is essentially what it is. The current `ProgressChart` component in my opinion should be renamed, to something like `ResourceProgress` or `ResourceCompletion`? 

Note that it also means that the module should be renamed. `progresschart` doesn't tell what progress we're looking at.